lib/rules/prefer-wait-for.ts

```diff
@@ -112,46 +150,100 @@ export default ESLintUtils.RuleCreator(getDocsUrl)<Options, MessageIds>({
     };
 
     return {
-      'ImportDeclaration[source.value=/testing-library/]'(
-        node: TSESTree.ImportDeclaration
-      ) {
-        const deprecatedImportSpecifiers = node.specifiers.filter(
-          (specifier) =>
-            isImportSpecifier(specifier) &&
-            specifier.imported &&
-            DEPRECATED_METHODS.includes(specifier.imported.name)
-        );
-
-        deprecatedImportSpecifiers.forEach((importSpecifier, i) => {
-          if (i === 0) {
-            reportImport(node);
-          }
-
-          context
-            .getDeclaredVariables(importSpecifier)
-            .forEach((variable) =>
-              variable.references.forEach((reference) =>
-                reportWait(reference.identifier)
-              )
-            );
-        });
+      'CallExpression > MemberExpression'(node: TSESTree.MemberExpression) {
+        const isDeprecatedMethod =
+          isIdentifier(node.property) &&
+          DEPRECATED_METHODS.includes(node.property.name);
+        if (!isDeprecatedMethod) {
+          // the method does not match a deprecated method
+          return;
+        }
+        const testingLibraryNode =
+          helpers.getCustomModuleImportNode() ??
+          helpers.getTestingLibraryImportNode();
+        // this verifies the owner of the MemberExpression is the same as the node if it was imported with "import * as TL from 'foo'"
+        const callerIsTestingLibraryFromImport =
+          isIdentifier(node.object) &&
+          isImportDeclaration(testingLibraryNode) &&
+          isImportNamespaceSpecifier(testingLibraryNode.specifiers[0]) &&
+          node.object.name === testingLibraryNode.specifiers[0].local.name;
+        // this verifies the owner of the MemberExpression is the same as the node if it was imported with "const tl = require('foo')"
+        const callerIsTestingLibraryFromRequire =
+          isIdentifier(node.object) &&
+          isCallExpression(testingLibraryNode) &&
+          isVariableDeclarator(testingLibraryNode.parent) &&
+          isIdentifier(testingLibraryNode.parent.id) &&
+          node.object.name === testingLibraryNode.parent.id.name;
+        if (
+          !callerIsTestingLibraryFromImport &&
+          !callerIsTestingLibraryFromRequire
+        ) {
+          // the method does not match from the imported elements from TL (even from custom)
+          return;
+        }
```

Comment:
Should we extract this into its own detection helper? A function which determines if a given node was directly imported from testing library (either default or custom module). It looks like we are gonna end up repeating this bit across different rules from now on.